Feature/794/gatling in concourse prod

VivianAllen opened the pull request:
Add concourse test to run glue daily and MI workflows at the same time
as a gatling test add 36 users / sec for 1200 seconds (resulting in
a constant 1008 users making 1008 req/s after 28 seconds)).

Test consists of two jobs (start workflows and run gatling) which
needed two additional tasks to build and run the gatling test with
maven and to get credentials to start the workflows and to upload
the gatling reports to s3.

The test is scheduled to run at 01:00

VivianAllen commented:
NB draft as @sanisc's work is included here (we branched off his work) and is yet to be merged, so this will probably need rebasing once that is in.

sanisc commented:
> NB draft as @sanisc's work is included here (we branched off his work) and is yet to be merged, so this will probably need rebasing once that is in.

this has been done now so feel free to rebase

sanisc commented:
Having talked about this at the cross team tech lead standup it was raised that the data entered should be removed once the test is complete. I have spoken @VivianAllen about this and there is a possibility that it could be a glue job. Any thoughts @samcrang from an implementation perspective?

sanisc commented on concourse/tasks/gatling-workflow-load-test.yml:
```diff
@@ -0,0 +1,36 @@
+        -DinjectDurationSeconds=1200 \
+        -DpauseBetweenRequestsInSeconds=1 \
+        -DnumberOfRepetitions=1 \
+        -Dgatling.simulationClass=svp.SvpSimulationConstantUsersPerSec \
```
i don't see the following scala class in the PR anymore SvpSimulationConstantUsersPerSec.scala?

VivianAllen replied:
ah balls, must have lost it in the rebase. I'll reinstate it, thanks!

VivianAllen replied:
done, thanks for the catch!

VivianAllen commented:
NB - draft still as @vijay-p-123 and I are working on the cleanup implemention

VivianAllen commented:
Hey @sanisc @samcrang @stephengrier @junaidsharif this is now ready for review

BolaOgunlana reviewed (commented):
An update to the (High level) README.md or a new related README.md file will certainly be of benefit to go along with this code. 
Some guidance on how to point the Gatling tests at a specific Environment would be of value (to those who only visit concourse occasionally)

VivianAllen commented:
> An update to the (High level) README.md or a new related README.md file will certainly be of benefit to go along with this code.
> Some guidance on how to point the Gatling tests at a specific Environment would be of value (to those who only visit concourse occasionally)

Good point, will do. Gatling cannot be pointed at anything other than staging at the moment, as its the only environment with a working instance of the front end

sanisc commented:
> > An update to the (High level) README.md or a new related README.md file will certainly be of benefit to go along with this code.
> > Some guidance on how to point the Gatling tests at a specific Environment would be of value (to those who only visit concourse occasionally)
> 
> Good point, will do. Gatling cannot be pointed at anything other than staging at the moment, as its the only environment with a working instance of the front end

a readme exists which i wrote here https://github.com/alphagov/govuk-shielded-vulnerable-people-service/tree/master/loadtests 

feel free to update that it has pretty much everything needed to run it or work it out

VivianAllen commented:
Hey, I've updated the README with some stuff about the invidual simulations, AWS environments, and concourse. @sanisc I've left the section on the svpSimulation blank as I wasn't sure if you would prefer to fill that in - happy to have a go myself however.

(I also fixed a bug where the extra config parameters we added hadn't been moved across from our dev branch properly)

sanisc commented on loadtests/src/test/main/scala/svp/SvpSimulationConstantUsersPerSec.scala:
```diff
@@ -0,0 +1,17 @@
+  setUp(ShieldedVulnerablePeople.scnEndToEndJourney.inject(
+    constantUsersPerSec(Config.injectUsersPerSecond) during (Config.injectDurationSeconds seconds)))
+    .protocols(Config.httpProtocol)
+    .maxDuration(Config.duration * 60)
```
this duration can be just in minutes now without the * 60

VivianAllen replied:
done

BolaOgunlana reviewed (approved):
All outstanding comments appear to have been addressed satisfactorily
